### ISSUE TITLE
Error overriding a method with an implicitly scoped parameter with -langversion:10

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#63761]
An error is reported overriding a method with an implicitly scoped parameter when the override is compiled with `-langversion:10`.

Compile base class with `-langversion:11`
```csharp
public abstract class A<T>
{
    public abstract void F(out T t);
}
```

Compile derived class with `-langversion:10` and .NET 6
```csharp
class B : A<int>
{
    // error CS8987: The 'scoped' modifier of parameter 'i' doesn't match overridden or implemented member.
    public override void F(out int i) { i = 0; }
}
```

Since `scoped` cannot be used with `-langversion:10`, perhaps the error should be a warning in this case.
